manuscript/chapter3.md

```diff
@@ -200,20 +278,32 @@ class App extends Component {
 
 Let's recap what happens during the component lifecycle. Your component gets initialized by the constructor. After that, it renders for the first time. But you prevent it from displaying anything, because the result in the local state is null. It is allowed to return null for a component in order to display nothing. Then the `componentDidMount()` lifecycle method runs. In that method you fetch the data from the Hacker News API asynchronously. Once the data arrives, it changes your internal component state in `setSearchTopStories()`. Afterward, the update lifecycle comes into play because the local state was updated. The component runs the `render()` method again, but this time with populated result in your internal component state. The component and thus the Table component with its content will be rendered.
 
+让我们回顾一下在整个组将的生命周期中发生了什么。首先组件通过构造函数得到初始化，之后它将初始化的状态渲染出来。但是当结果为空时组件返回空，这阻止了组件的渲染。React允许组件通过返回 null 来不渲染任何东西。接着 `componentDidMount()` 生命周期函数执行。在这个方法中你从 Hacker News API 中异步地拿到了数据。一旦数据到达，组件就通过 `setSearchTopStories()` 函数改变组件内部的状态。之后，因为状态的更新，更新生命周期开始运行。组件再次执行 `render()` 方法，但这一次内部组件的状态中的结果已经被填充不再为 null。因此组件将渲染 Table 组件的内容。
```

Comment:
让我们回顾一下在整个组将的生命周期中发生了什么。
-> 
让我们回顾一下在组件的整个生命周期中发生了什么。

Reply:
但是当结果为空时组件返回空，这阻止了组件的渲染。
->
但是你阻止组件什么都去显示，因为此时本地状态中的结果为空。

Reply:
但这一次内部组件的状态中的结果已经被填充不再为 null。
-> 
但这次组件的内部状态中的结果已经填充，不再是空了。

Reply:
*但是当结果为空时组件返回空，这阻止了组件的渲染。 *
改成 *但是你阻止组件什么都去显示，因为此时本地状态中的结果为空。* 我觉得不好，根据代码来看，首先是判断结果是否为null，然后组件才决定是否渲染。所以当时就是这样翻译的。如果改成这样感觉本末倒置了。

Reply:
But you prevent it from displaying anything, because the result in the local state is null.

我理解这句的原句意思更多是强调作为写代码的人，去阻止了组件渲染空结果的这个行为，而不是组件自己决定自己的行为。我自己理解是作者希望读者能够想到要去处理结果为空的情况，所以我就还是把“你”提出来当主语了。

Reply:
嗯，这么一说我觉得有道理。
改为 -> 但是你阻止了任何组件的显示，因为此时本地状态中的结果为空。 
怎么样？